repos/system_upgrade/el7toel8/actors/checkkdegnome/libraries/library.py

```diff
@@ -0,0 +1,66 @@
+from leapp.models import InstalledDesktopsFacts, InstalledKdeAppsFacts
+
+
+def check_kde_gnome():
+    desktopFacts = next(api.consume(InstalledDesktopsFacts))
+    kde_desktop_installed = desktopFacts.kde_installed
+    gnome_desktop_installed = desktopFacts.gnome_installed
+
+    # No desktop installed, we don't even care about apps as they are most likely not used or even installed
+    if not kde_desktop_installed and not gnome_desktop_installed:
+        api.current_logger().info("No desktop installed. Continuing with the upgrade.")
+        return
+
+    if kde_desktop_installed:
+        api.current_logger().info("KDE desktop is installed. Checking what we can do about it.")
+        if not gnome_desktop_installed:
+            api.current_logger().error("Cannot perform the upgrade because there is"
+                                       " no other desktop than KDE installed.")
+            # We cannot continue with the upgrade process
+            reporting.create_report([
+                reporting.Title("Cannot upgrade because there is no other desktop than KDE installed."),
+                reporting.Summary("With only KDE installed, there would be no other desktop env. after upgrade."),
+                reporting.Severity(reporting.Severity.HIGH),
+                reporting.Tags([
+                    reporting.Tags.UPGRADE_PROCESS
+                ]),
+                reporting.Flags([
+                    reporting.Flags.INHIBITOR
+                ]),
+                reporting.Remediation(
+                    hint="Install GNOME desktop to be able to upgrade.")
+                ])
+            return
+
+        # Assume both GNOME and KDE are installed in this state
+        api.current_logger().info("Upgrade can be performed, but KDE desktop will"
+                                  " be removed in favor of GNOME")
+        reporting.create_report([
+            reporting.Title("Upgrade can be performed, but KDE will be uninstalled."),
+            reporting.Summary("KDE has to be uninstalled in favor of GNOME to perform the upgrade."),
+            reporting.Severity(reporting.Severity.MEDIUM),
+            reporting.Tags([
+                reporting.Tags.UPGRADE_PROCESS
+            ])])
+        api.current_logger().info("----------------------------------")
+
+    # At this state we just need to detect whether any KDE/Qt app is installed to inform user
+    # that the application will be removed during the upgrade process. No matter if KDE is installed
+    # or not.
+
+    KDEAppsFacts = next(api.consume(InstalledKdeAppsFacts))
+    if KDEAppsFacts.installed_apps:
+        # upgrade can be performed, but user will loose KDE apps
+        api.current_logger().info("Installed KDE/Qt apps detected.")
+        reporting.create_report([
+            reporting.Title("Upgrade can be performed, but KDE/Qt apps will be uninstalled."),
+            reporting.Summary("KDE/Qt apps will be removed to perform the upgrade."),
```

Comment:
```suggestion
            reporting.Summary("The KDE desktop environment is not available on RHEL 8. All the KDE/Qt apps will be removed during the upgrade, including but not limited to:\n- {0}".format("\n- ".join(KDEAppsFacts.installed_apps))),
```
I think the current message is not sufficient. It's just too generic:
```
Risk Factor: medium
Title: Upgrade can be performed, but KDE/Qt apps will be uninstalled.
Summary: KDE/Qt apps will be removed to perform the upgrade.
```
Please apply the coding standards to the proposed change (max 120 chars one line).